src/Compilers/Core/MSBuildTaskTests/TestUtilities/DotNetSdkTestBase.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.BuildTasks.UnitTests
+{
+    public abstract partial class DotNetSdkTestBase : TestBase
+    {
+        public sealed class DotNetSdkAvailable : ExecutionCondition
+        {
+            public override bool ShouldSkip => s_dotnetSdkPath == null;
+            public override string SkipReason => "The location of dotnet SDK can't be determined (DOTNET_INSTALL_DIR environment variable is unset)";
+        }
+
+        private static readonly string s_dotnetExeName;
+        private static readonly string s_dotnetInstallDir;
+        private static readonly string s_dotnetSdkVersion;
+        private static readonly string s_dotnetSdkPath;
+
+        private static string s_projectSource =
+@"<Project Sdk='Microsoft.NET.Sdk'>
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+  </PropertyGroup>
+</Project>
+";
+        private static string s_classSource =
+@"using System;
+
+public class TestClass 
+{
+    public void F() 
+    {
+        Console.WriteLine(123);
+    }
+}
+";
+
+        protected readonly TempDirectory ProjectDir;
+        protected readonly TempDirectory ObjDir;
+        protected readonly TempDirectory OutDir;
+        protected readonly TempFile Project;
+        protected readonly string ProjectName;
+        protected readonly string ProjectFileName;
+        protected readonly string Configuration;
+        protected readonly string TargetFramework;
+        protected readonly string DotNetPath;
+        protected readonly IReadOnlyDictionary<string, string> EnvironmentVariables;
+
+        private int _logIndex;
+
+        private static string GetSdkPath(string dotnetInstallDir, string version)
```

Comment:
Perhaps make this a local function, like `IsMatchingDotNetInstance` since only used in the static ctor?

Reply:
I'd need to invent new name for `dotnetInstallDir` then, since it's already used. Kind of a drawback of local functions...